File: dto-mapping-benchmark/src/main/java/com/example/dtomappingbenchmark/v4/service/PostService.java
package com.example.dtomappingbenchmark.v4.service;

import com.example.dtomappingbenchmark.v4.dto.PostDto;
import com.example.dtomappingbenchmark.v4.entity.Post;
import com.example.dtomappingbenchmark.v4.entity.User;
import com.example.dtomappingbenchmark.v4.mapper.PostMapper;
import com.example.dtomappingbenchmark.v4.repository.PostRepository;
import com.example.dtomappingbenchmark.v4.repository.UserRepository;
import lombok.RequiredArgsConstructor;
import org.springframework.stereotype.Service;

@Service
@RequiredArgsConstructor
public class PostService {
    private final PostRepository postRepository;
    private final UserRepository userRepository;

    public PostDto create(PostDto dto){
        //매퍼 변환
        Post post = PostMapper.toEntity(dto);
        User user = userRepository.findById(dto.getUserId()).orElseThrow();
        PostMapper.patchUser(user, post);
        return PostMapper.toDto(postRepository.save(post));
    }
}

File: object-mapping-benchmark-v4/src/main/kotlin/yu/selab/cj/objectmappingbenchmarkv4/post/service/PostService.kt
package yu.selab.cj.objectmappingbenchmarkv4.post.service

import org.springframework.http.RequestEntity.post
import org.springframework.stereotype.Service
import org.springframework.transaction.annotation.Transactional
import yu.selab.cj.objectmappingbenchmarkv4.mapper.PostMapper
import yu.selab.cj.objectmappingbenchmarkv4.post.command.CreatePostDto
import yu.selab.cj.objectmappingbenchmarkv4.post.command.ReadPostDto
import yu.selab.cj.objectmappingbenchmarkv4.post.repository.PostRepository
import yu.selab.cj.objectmappingbenchmarkv4.user.repository.UserRepository
import kotlin.collections.map


@Service
class PostService(
    private val postRepository : PostRepository,
    private val userRepository : UserRepository
) {

    @Transactional(readOnly = true)
    fun getPostList() = postRepository.findAll().map{
//        post -> ReadPostDto(null, null, null).fromEntity(post)
        post -> PostMapper.toDto(post)
    }

    @Transactional(readOnly = true)
    fun getPost(id : Long) : ReadPostDto {
        val post = postRepository.findById(id).orElseThrow {
            IllegalArgumentException("게시물 없음")
        }

        return PostMapper.toDto(post)
    }

    @Transactional
    fun createPost(dto : CreatePostDto) {
        val user = userRepository.findById(dto.userId).orElseThrow {
            IllegalArgumentException("유저 없음")
        }
        val post = PostMapper.toEntity(dto, user)
        postRepository.save(post)

//        postRepository.save(Post(dto.title, dto.content, user))
    }

    @Transactional
    fun deletePost(id : Long) {
        postRepository.deleteById(id)
    }

}